### PR TITLE
{Storage-preview} Secured SMB

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage-preview

File: src/storage-preview/azext_storage_preview/_params.py [hunk 1]
```diff
@@ -127,8 +127,22 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+                   min_api='2020-08-01-preview',
+                   help="SMB authentication methods supported by server. Valid values are NTLMv2, Kerberos. "
+                        "Should be passed as a string with delimiter ';'.")
+        c.argument('kerberos_ticket_encryption', options_list=['--kerb-ticket-encryption', '-k'],
+                   arg_group='SMB Setting', min_api='2020-08-01-preview',
+                   help="Kerberos ticket encryption supported by server. Valid values are RC4-HMAC, AES-256. "
+                        "Should be passed as a string with delimiter ';'.")
+        c.argument('channel_encryption', arg_group='SMB Setting', min_api='2020-08-01-preview',
+                   help="SMB channel encryption supported by server. Valid values are AES-CCM-128, AES-GCM-128, "
+                        "AES-GCM-256. Should be passed as a string with delimiter ';'.")
```
Comment (evelyn-ys) [inline src/storage-preview/azext_storage_preview/_params.py:145]: Do these valid values have enum definition in SDK? Can we add format validator for these `SMB Setting`?

Comment (Juliehzl) [reply src/storage-preview/azext_storage_preview/_params.py:145]: No. They are not one specific value. The accepted value is a combination of several valid values with `,`.

Comment (evelyn-ys) [reply src/storage-preview/azext_storage_preview/_params.py:145]: I see. By format validator, I mean splitting user input and check whether each value is valid(within enum options). But if there's no enum in SDK, then fine with me.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**